### ISSUE TITLE
DockerDesktop don't work after 4.4.2 upgrade to 4.5.0 for my m1pro mac

### ISSUE THREAD

**Issue by jeyrce** [docker/for-mac#6175]
### main information

- macos 12.1
- m1 chip
- from 4.4.2(worked) upgrade to 4.5.0(not work)

While I uninstall the not worked version 4.5.0, and then install 4.5.0 directly, it works. Seems there is something wrong in upgrading

**Comment by posita** [comment]
The upgrade bricked my install as well. (I'm on Intel.)

![screenshot](https://user-images.githubusercontent.com/222581/154817342-86670c05-4ee4-4505-8f6d-33c5ddc024bd.png)

``` sh
% docker ps --all
zsh: command not found: docker
```

I had to download the 4.5.0 `.dmg` and manually reinstall.

**Comment by alexkoepke** [comment]
I had the the same problem Upgrading from Docker Desktop 4.3.2 only thing that solved it was downloading the .dmg and reinstalling. 
MacBook Pro (Retina, 15-inch, Mid 2014)

<img width="698" alt="image" src="https://user-images.githubusercontent.com/4930741/154820032-0b6c8fbd-0624-4d6d-bd23-9ed20933c234.png">

**Comment by sha1sum** [comment]
I had this very same issue. I clicked to restart and update and Docker quit and then... nothing. I searched Spotlight for "Docker" and "Docker.app" showed with no icon, and when I attempted to start it I saw a notification that the app is damaged and should be moved to the trash. Upon trying to reinstall it from the dmg download I saw the following:

![image](https://user-images.githubusercontent.com/7863437/156465048-aa3e6e92-8e7f-4459-b475-5dce51f565d2.png)

The solution for me turned out to be moving it to the trash _and then restarting the machine after_ before attempting to re-drag the app from the dmg into Applications (which then worked).
